scripts/import_models_from_detectron.sh

```diff
@@ -0,0 +1,205 @@
+    readme_data=""
+
+    # Check if model is supported and exit if it isn't
+    if [ "${model}" == "e2e_faster_rcnn_R-50-C4_1x" ]; then
+        # Include model id number at the end
+        id="35857197"
+        base_name="${model}_${id}"
+        net_name="detectron_${base_name}"
+        pkl_name="${base_name}.pkl"
+        pkl_url="${DETECTRON_S3_ROOT}35857197/12_2017_baselines/e2e_faster_rcnn_R-50-C4_1x.yaml.01_33_49.iAX0mXvW/output/train/coco_2014_train%3Acoco_2014_valminusminival/generalized_rcnn/model_final.pkl"
+        cfg="configs/12_2017_baselines/e2e_faster_rcnn_R-50-C4_1x.yaml"
+        # We can't reliably extract this from the model, so specified here
+        value_info_data='{"data": [1, [1, 3, 800, 800]]}'
+        readme_data="# Detectron ${model}
+
+* Original: https://github.com/facebookresearch/Detectron/blob/master/MODEL_ZOO.md#end-to-end-faster--mask-r-cnn-baselines
+* Backbone: R-50-C4
+* Type: Faster
+* LR Schedule: 1x
+* Model ID: ${id}
+
+### Install
+
+To install this model, run
+
+    python -m caffe2.python.models.download -i detectron/${model}
+
+"
+    else
+        echo "Detectron model ${model} isn't supported yet. Exiting."
+        exit 1
+    fi
+
+    pushd "${detectron_dir}"
+    mkdir -p "models/pb_output"
+
+    # Download the pkl file
+    if [ ! -f "models/${pkl_name}" ]; then
+        echo "Downloading ${model}..."
+        wget "${pkl_url}" -O "models/${pkl_name}"
+    fi
+
+    # Convert for CPU and run a test image
+    echo "Converting ${model}..."
+    python tools/convert_pkl_to_pb.py --use_nnpack 0 --net_name "${net_name}" --cfg "${cfg}" --out_dir "models/pb_output/${model}/" --test_img "demo/16004479832_a748d55f21_k.jpg" TEST.WEIGHTS "models/${pkl_name}"
+
+    # Convert for NNPACK and *don't* pass in a test image, since this
+    # likely won't work on a server machine
+    # python tools/convert_pkl_to_pb.py --use_nnpack 1 --net_name "${net_name}" --cfg "${cfg}" --out_dir "models/pb_output/${model}_nnpack/" TEST.WEIGHTS "models/${pkl_name}"
+
+    # Switch back to Caffe2 models directory
+    popd
+
+    # Copy the model files over
+    if [ ${dry_run} -eq 0 ]; then
+        mkdir -p "detectron/${model}"
+        cp "${detectron_dir}/models/pb_output/${model}/model_init.pb" "detectron/${model}/init_net.pb"
+        cp "${detectron_dir}/models/pb_output/${model}/model.pb" "detectron/${model}/predict_net.pb"
+        cp "${detectron_dir}/models/pb_output/${model}/model.pbtxt" "detectron/${model}/predict_net.pbtxt"
+        cp "${detectron_dir}/models/pb_output/${model}/model_def.png" "detectron/${model}/predict_net_def.png"
+        cp "${detectron_dir}/models/pb_output/${model}/model_def.png" "detectron/${model}/predict_net_def.png"
+        echo "${value_info_data}" > "detectron/${model}/value_info.json"
+        echo "${readme_data}" > "detectron/${model}/README.md"
+    fi
+
+    echo ""
+}
+
+# Detectron requires the build directory in our path
+if [ ! -d "${caffe2_dir}/build" ]; then
```

Comment:
since we use pip uninstall, shall we also ask people to use pip install or python setup.py develop?

Reply:
You're right - I'll change this to pip install

Reply:
Unfortunately it looks like `pip install` and `python setup.py` both exclusively use `.setuptools-cmake-build` as the build directory and I could only get Detectron to work when we use `build` as the target directory. At this point I'm going to land it but will talk with @bddppq tomorrow about getting Detectron comfortable with a variety of install methods.